### ISSUE TITLE
rule: table-no-margin

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#123]
## Rule ID
`table-no-margin`

## What does this rule detect?
Margins do not apply to internal table elements (`table-row`, `table-row-group`, `table-header-group`, `table-footer-group`, `table-column`, `table-column-group`). Only `table`, `inline-table`, and `table-caption` support margins.

## Example: should warn
```html
<table>
  <tr style="margin: 10px;"><td>cell</td></tr>
</table>
```

## Example: should NOT warn
```html
<table style="margin: 10px;">
  <tr><td>cell</td></tr>
</table>
```

## Related CSS Properties
`display`, `margin`, `margin-top`, `margin-right`, `margin-bottom`, `margin-left` (and logical equivalents)

## References
- [CSS 2.2 §17.5 - Visual layout of table contents](https://www.w3.org/TR/CSS22/tables.html#table-display): margins do not apply to internal table elements
- [Firefox inactive-property-helper.js](https://searchfox.org/mozilla-central/source/devtools/server/actors/utils/inactive-property-helper.js) — detects this pattern
- Extremely common mistake (margin on `<tr>`)
